docs/develop/smart-contracts/guidelines/message-delivery-guarantees.mdx

```diff
@@ -0,0 +1,70 @@
+
+Moreover if account A sent two messages to account B, it is guaranteed that message with lower _lt_ will be processed earlier. Otherwise, an attempt to synchronize delivery would require the state of all the others to be known before processing one shard, and thereby break parallelization and destroy efficient sharding.
+
+
+For each block, we can define its _lt_ span, which starts with the first transaction and ends with the _lt_ of the last event in the block (message or transaction). Blocks are ordered the same way as any other events in TON, that way if one block depends on the other it has higher _lt_. Child block in shard has higher _lt_ than it's parent. Masterchain block _lt_ is higher that _lt's_ of shard blocks it lists, since master block _depends_ on listed shardblocks. In order each shard block contains reference to latest (at the moment of shard block creation) masterblock and thus shard block _lt_ is higher than referenced master block _lt_.
+
+## Message delivery
+
+Fortunately, TON is working such way that any internal message will surely be received on the destination account. Message cannot be lost anywhere between the source and destination. External messages is a little bit different since it's acceptance to the block is at validators discretionm however once message is accepted to the incoming message queue it will be delivered.
+
+### Delivery order
+
+It looks like _lt_ solves the issue about message delivery order, because we know that transaction with lower _lt_ will be processed first. But doesn't work for all scenarios.
+
+Suppose that there are two contracts - _A_ and _B_. Then _A_ receives an external message which triggers it to send two internal messages to _B_, let's call these messages _1_ and _2_. In that simple case we can be 100% sure that _1_ will be processed by _B_ before _2_ because it has lower _lt_.
+
+<ConceptImage src="/img/docs/msg-delivery-1.png" />
+
+It is just a simple case when we have only two contracts. What if our system works with more of them?
+
+### Several smart contracts
+
+Suppose that we have three contracts - _A_, _B_ and _C_. In some transaction, _A_ sends two internal messages _1_ and _2_: one to _B_ and another one to _C_. Even though they were created in some exact order (_1_, then _2_), we can't be sure that _1_ will be processed before _2_. That happens because routes from _A_ to _B_ and from _A_ to _C_ can differ in length and validator sets. If these contracts are in different shardchains, one of the messages may require several blocks to reach the destination contract.
```

Comment:
Suppose that we have three contracts - _A_, _B_ and _C_. In **a** transaction, _A_ sends two internal messages _1_ and _2_: one to _B_ and another to _C_. Even though they were created in **an** exact order (_1_, then _2_), we can't be sure that _1_ will be processed before _2_. **This is the case** because routes from _A_ to _B_ and from _A_ to _C_ can differ in length and validator sets. If these contracts are in different **shard chains**, one of the messages may require several blocks to reach the destination contract.